pkg/pma/pma.h

```diff
@@ -0,0 +1,169 @@
+/// @file
+///
+///
+
+#ifndef PMA_PMA_H
+#define PMA_PMA_H
+
+#include <stddef.h>
+#include <stdint.h>
+
+//==============================================================================
+// TYPES
+
+/// Status of a page in a PMA.
+typedef enum page_status {
+    /// Page hasn't been read or written.
+    PS_UNMAPPED = 0x0,
+    /// Page has been read but not written.
+    PS_MAPPED_CLEAN = 0x1,
+    /// Page has been written.
+    PS_MAPPED_DIRTY = 0x2,
+    /// Page is mapped but can't be read or written.
+    PS_MAPPED_INACCESSIBLE = 0x3,
+    PS_MASK                = 0x3,
+} page_status_t;
```

Comment:
Should probably just leave as is, but some trickery could get you the length as a function of the number of members:

```c
   page_status {
     PS_UNMAPPED,
     ...,
     PS_MAPPED_INACCESSIBLE,
     PS_NMEMB,
     PS_MASK = 0x3 // or you could write a compile time macro to round PS_NMEMB up to power of two-1 ;) -- probably don't do this
   };
```

Reply:
And then, `kBitsPerPage = PS_NMEMB` -- now more than just a comment referencing `page_status_t`. IDK, this is just a pattern i like.